app/src/main/java/me/quadphase/qpdex/AdvancedSearch.java

```diff
@@ -0,0 +1,512 @@
+package me.quadphase.qpdex;
+
+import android.app.Activity;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.inputmethod.EditorInfo;
+import android.widget.Spinner;
+import android.content.Intent;
+import android.view.KeyEvent;
+
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.TextView;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import me.quadphase.qpdex.databaseAccess.PokemonFactory;
+import me.quadphase.qpdex.pokemon.Ability;
+import me.quadphase.qpdex.pokemon.EggGroup;
+import me.quadphase.qpdex.pokemon.MinimalPokemon;
+import me.quadphase.qpdex.pokemon.Type;
+
+
+/**
+ * Created by Thinesh on 31-Aug-15
+ *
+ * Class deals with the advanced search feature accessed from main menu
+ *
+ * Can filter pokemon by
+ *
+ * Type 1
+ * Type 2
+ * Ability
+ * Generation Available
+ * Egg Groups 1
+ * Egg Groups 2
+ * TODO: All 6 Base stats
+ * TODO: The ability to include/exclude Primal/Mega/Alternate Forms
+ *
+ */
+
+
+public class AdvancedSearch extends ActionBarActivity {

```

Comment:
Refactor this to "AdvancedSearchActivity extends AppCompatActivity". ActionBarActivity is now deprecated.